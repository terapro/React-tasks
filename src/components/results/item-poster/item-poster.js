import React from 'react';

import PropTypes from 'prop-types';

const ItemPoster = ({posterLink, posterClick, id}) => (
    <div className='item-poster'>
      <img src={posterLink} className='item-poster-img' onClick={posterClick} id={id}/>
    </div>
  );


ItemPoster.propTypes= {
  posterLink: PropTypes.string.isRequired,
  posterClick: PropTypes.func,
  id: PropTypes.string
};

ItemPoster.defaultProps = {
  posterClick: () => {},
  id: ''
};
export {ItemPoster};