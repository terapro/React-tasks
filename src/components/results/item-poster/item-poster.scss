$posterHeight: 300px;
$posterWidth: 200px;

.item-poster {
  width: $posterWidth;
  height: $posterHeight;
  cursor: pointer;
}

.item-poster-img {
  width: $posterWidth;
  height: $posterHeight;
}