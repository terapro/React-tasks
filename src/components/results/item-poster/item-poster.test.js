import React from 'react';
import {ItemPoster} from 'src/components/results/item-poster/item-poster.js';
import {shallow} from 'enzyme';


const mockLinkText ='https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg';
const mockFunc = jest.fn();
const mockIdtext = 'id03'

describe('<ItemPoster />', () => {
  describe('Rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(
        <ItemPoster posterLink={mockLinkText}
                    posterClick = {mockFunc}
                    id = {mockIdtext}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Behavior', () => {
    it('invokes onClick', () => {
      shallow(<ItemPoster posterClick={mockFunc} id={mockIdtext} />)
        .find('img#' + mockIdtext)
        .simulate('click');
      expect(mockFunc).toHaveBeenCalled();
    });
  });


});
