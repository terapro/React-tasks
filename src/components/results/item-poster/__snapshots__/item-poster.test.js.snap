// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemPoster /> Rendering should render the component 1`] = `
<div
  className="item-poster"
>
  <img
    className="item-poster-img"
    id="id03"
    onClick={[MockFunction]}
    src="https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg"
  />
</div>
`;
